### DISCUSSION TITLE
How do I convert the with-apollo example to Next 13 app directory?

### DISCUSSION THREAD

**Discussion by jetaggart** [vercel/next.js#45917] [Help]
Hello, I'm struggling to convert the with-apollo example to next 13 app directory. Specifically, how do I convert the `_app.jsx` to `layout.jsx` when the `_app.jsx` is pulling from pageProps (what is the type of pageProps?)

[As seen here](https://github.com/vercel/next.js/blob/a485dc9af197d7da30ffc02c212fab278b28696c/examples/with-apollo/pages/_app.js#L4-L5):
```javascript
import { ApolloProvider } from '@apollo/client'
import { useApollo } from '../lib/apolloClient'

export default function App({ Component, pageProps }) {
  const apolloClient = useApollo(pageProps)

  return (
    <ApolloProvider client={apolloClient}>
      <Component {...pageProps} />
    </ApolloProvider>
  )
}
```

How do I convert that to the new app directory?

**Comment by ciruz** [comment]
Sorry I'm on mobile so I can't give a really detailed answer.

First check this out, how to use Providers in the new RootLayout, this is basically the replacement for the old app file.

https://beta.nextjs.org/docs/rendering/server-and-client-components#rendering-third-party-context-providers-in-server-components

You can then use Apollo like so, by passing the client and children.

https://www.apollographql.com/docs/react/get-started/#step-4-connect-your-client-to-react

**Reply by jetaggart** [reply]
Thanks, I'll check those out. I think the root issue is I still have no way to hook into the page props to hydrate the apollo cache, since page props don't really exist anymore.

FWIW, this looks like a much larger undertaking. For reference:

https://github.com/apollographql/apollo-client/issues/10466
https://github.com/apollographql/apollo-client/issues/10344
https://github.com/apollographql/apollo-client/issues/10392
